core/common/src/annotation/TextAnnotation.ts

```diff
@@ -65,6 +65,45 @@ export interface TextFrameStyleProps {
   borderWeight?: number;
 }
 
+/**
+ * Specifies where to attach a leader line to a [[TextAnnotation]].
+ * The leader line can be attached to a key point, to the first or last line of text, or to the nearest point on the text block frame.
+ * @beta
+ */
+export type LeaderAttachmentMode =
+  | { mode: "KeyPoint"; curveIndex: number; fraction: number }
+  | { mode: "TextPoint"; position: LeaderTextPointOptions }
+  | { mode: "Nearest" };
+
+/**
+ * Specifies the possible positions for the leader attachment point relative to the annotation when attachmentMode is TextPoint.
```

Comment:
I assume the leader attaches to the annotation's frame. How does this work when the frame is not rectangular? e.g., what does `TopLeft` mean if the frame is a triangle?